### ISSUE TITLE
Code signing, certificates, intellisense xll

### ISSUE THREAD

**Issue by nVenticDW** [Excel-DNA/IntelliSense#62]
I've noticed that no intellisense dll's are signed. If I make intellisense available to users and signed xll are required by their IT administration would you make signed versions available (at least with SHA1) or  would you like me to sign the code with my company certificate for my users, noting the original source. Or do we just don't make (install) intellisense for such users.
Maybe a signed version of intellisense should be available at some cost since you need to cover the certificate authority fees and then you could also charge for some development. Intellisense is a highly valuable utility and needs continual care given microsoft updates.

After having looked at licensed xll and excel addins I have, I have noticed that most excel developers don't sign their code. As security clamps down over the years this might become more difficult, yet again it might remain as is.

**Comment by govert** [comment]
We need to distinguish between signing the .dll library which may be incorporated into a user's add-in, and signing the packaged .xll add-ins (either yours or the pre-built `ExcelDna.IntelliSense.xll`).

* Signing the `ExcelDna.IntelliSense.dll` itself serves no purpose.

* For you to sign your compiled and packed .xll add-in which may incorporate `ExcelDna.IntelliSense.dll` is a good idea. Excel will appropriately enforce whatever security is related to this.

* Making a signed pre-built version of the `ExcelDna.IntelliSense.xll` add-in is also a good idea. You're the first to ask, though. So I'm not sure of the urgency or the 'market' for it, and I'll wait and see a bit.

**Comment by govert** [comment]
Yes - I agree that signing the .xll is useful. If you're going to sign it with your own certificate, I'd prefer that you rename the .xll (just to prevent future confusion). You can rename the .xll add-in to anything you want without any problem. `nVecticDW.IntelliSense.xll` perhaps?

I don't understand that part about "compile to use strong certification". VS 2012 won't compile the project. Install VS 2017 and everything should compile without problems. But compiling and adding a certificate to the .dll (using the .NET code signing tools `sn.exe` for making strong named .NET assemblies) is not effective for anything in this context. 

You just need to use `signtool.exe` on the compiled and packed native code .xll as you're correctly showing in your images.

**Comment by govert** [comment]
I think we were talking past each other in the earlier posts, if you're not a .NET developer.

For .NET add-ins there is a way to incorporate a library called `ExcelDna.IntelliSense.dll` into the user's project. Signing this file is pointless - users should sign the .xll add-in they build instead.

You are using `ExcelDna.IntelliSense.xll` which is the independent add-in version. Signing this is valuable as you show, just not something I've done yet.
